### ISSUE TITLE
Quarkus Rest client can't consume /health and /metrics properly

### ISSUE THREAD

**Issue by pjgg** [quarkusio/quarkus#14996]
## Description

microprofile-rest-client delivered  can't consume /health and /metrics properly, because doesn't follow redirects. 

### Steps to reproduce

Add  Quarkus Rest client as a dependency:

```
<dependency>
    <groupId>io.quarkus</groupId>
    <artifactId>quarkus-rest-client</artifactId>
</dependency>
```

Setup a Rest client:

```
import org.eclipse.microprofile.rest.client.inject.RegisterRestClient;

import javax.ws.rs.GET;
import javax.ws.rs.Path;
import javax.ws.rs.core.Response;

@Path("/api")
@RegisterRestClient
public interface HealthClientService {

    @GET
    @Path("/health")
    Response health();
}

```

Properties:

```
quarkus.http.root-path=/api
quarkus.http.port=8081

io.quarkus.ts.openshift.http.HealthClientService/mp-rest/url=http://localhost:8081
io.quarkus.ts.openshift.http.HealthClientService/mp-rest/scope=javax.inject.Singleton
```

And then make a query:

```
@Inject
@RestClient
private HealthClientService healthClient;
 
assertThat(200, equalTo(healthClient.health().getStatus()));
```

You will get an Http status 301

**Comment by geoand** [comment]
Isn't there any config setting to make this work @michalszynkiewicz ?

**Comment by michalszynkiewicz** [comment]
Sadly, there is no MicroProfile Rest Client setting to make it work, it's been introduced in MP RC 2.0: https://github.com/eclipse/microprofile-rest-client/pull/274 and we're on 1.4.x.

There should be a way to set it through RestEasy settings. I found this in the docs but I'm not sure how to do it with MP API.

@ronsigal @asoldano maybe you know? Is there some some property mabe to enable following redirects?

**Comment by michalszynkiewicz** [comment]
Anyway, the endpoints are now (by default?) under `/q/health` if I understand correctly, so you should be able to this endpoint.
Right @kenfinnigan ?

**Comment by kenfinnigan** [comment]
Right, the solution would be to change the `@Path` in the REST Client interface

**Comment by jorsol** [comment]
Something must have changed on the rest-client, on 1.11.x the redirections were working, but on 1.12.0 it doesn't anymore.

And in my case, it's not the /health and /metrics endpoints, it's an endpoint for an external service.

**Comment by michalszynkiewicz** [comment]
@jorsol I spent some time on it recently and haven't found a way to make MP Rest Client, in the version that Quarkus currently uses, follow redirects.

As a work around you can use the RestEasy client directly and configure it to follow redirects:
https://github.com/michalszynkiewicz/quarkus-status/blob/flaky-tests/src/main/java/io/quarkus/status/flaky/GithubFlakyTestsProvider.java#L63-L67

**Comment by jorsol** [comment]
@michalszynkiewicz I have been hit by this change in RESTEasy: https://github.com/resteasy/Resteasy/pull/2623

Following the [documentation](https://docs.jboss.org/resteasy/docs/4.5.9.Final/userguide/html/ExceptionHandling.html#ResteasyWebApplicationException), the redirect from the client now throws an exception, so, catching the exception and unwrap the response to return the location Response.

So it's not really related to this issue, but still is a kind of breaking change in RESTEasy.

**Comment by ch3rub1** [comment]
I also encountered the redirect issue not for /health and /metrics endpoints but for an external service which respond 301.  
I found the same workaround than @michalszynkiewicz to resolve this issue temporarily.  
This issue seems also related to [this one](https://github.com/quarkusio/quarkus/issues/12517).
Resteasy 4.6.0.Final uses Microprofile Rest Client 2.0 which allow configuring Rest Clients to follow redirects (see [here](https://download.eclipse.org/microprofile/microprofile-rest-client-2.0-RC2/microprofile-rest-client-2.0-RC2.html#_following_redirect_responses)).  
Since I don't have any ideas of the impacts I may ask a silly question but does the upgrade to Resteasy 4.6.0.Final and Microprofile Rest Client 2.0 (by transitivity) can be considered ?
